x-pack/plugins/maps/public/components/map/feature_tooltip.js

```diff
@@ -11,52 +11,102 @@ import { i18n } from '@kbn/i18n';
 
 export class FeatureTooltip extends React.Component {
 
+  _renderFilterButton(tooltipProperty, hasFilters) {
+    if (!this.props.isFilterable || !hasFilters)  {
+      return null;
+    }
 
-  _renderProperties() {
-    return Object.keys(this.props.properties).map(propertyName => {
+
+    let icon;
+    if (tooltipProperty.isFilterable()) {
+      icon = (<EuiButtonIcon
+        iconType="plusInCircle"
+        title="Filter on property"
+        onClick={() => {
+          this.props.closeTooltip();
+          const filterAction = tooltipProperty.getFilterAction();
+          filterAction();
+        }}
+        aria-label="Filter on property"
+        className="mapFeatureTooltipFilterButton"
+      />);
+    } else {
+      icon = null;
+    }
+
+
+    return (
+      <EuiFlexItem className="mapFeatureTooltipFilterButton" grow={false}>
+        {icon}
+      </EuiFlexItem>
+    );
+  }
+
+  _renderProperties(hasFilters) {
+    return this.props.properties.map((tooltipProperty, index) => {
       /*
        * Justification for dangerouslySetInnerHTML:
-       * Propery value contains value generated by Field formatter
+       * Property value contains value generated by Field formatter
        * Since these formatters produce raw HTML, this component needs to be able to render them as-is, relying
        * on the field formatter to only produce safe HTML.
        */
-
       const htmlValue = (<span
         // eslint-disable-next-line react/no-danger
         dangerouslySetInnerHTML={{
-          __html: this.props.properties[propertyName]
+          __html: tooltipProperty.getHtmlDisplayValue()
         }}
       />);
+
       return (
-        <div key={propertyName}>
-          <span><strong>{propertyName}</strong></span>
-          <span>{' '}</span>
-          {htmlValue}
-        </div>
+        <EuiFlexGroup key={index}>
+          <EuiFlexItem grow={4}>
+            <strong>{tooltipProperty.getPropertyName()}</strong>
+          </EuiFlexItem>
+          <EuiFlexItem grow={6}>
+            {htmlValue}
+          </EuiFlexItem>
+          {this._renderFilterButton(tooltipProperty, hasFilters)}
+        </EuiFlexGroup>
       );
     });
   }
 
+  _renderCloseButton() {
+    if (!this.props.showCloseButton) {
+      return null;
+    }
+    return (
+      <EuiFlexGroup direction="column" gutterSize="none">
+        <EuiFlexItem grow={true}>
+          <EuiFlexGroup alignItems="flexEnd" direction="row" justifyContent="spaceBetween">
+            <EuiFlexItem>&nbsp;</EuiFlexItem>
```

Comment:
To get the effect you're going for, you don't need the empty flex item. You can just change `justifyContent` on the flex group to `flexEnd`